Feature: download selection of responses

jakobsitory opened the pull request:
<!-- We require pull request titles to follow the Conventional Commits specification ( https://www.conventionalcommits.org/en/v1.0.0/#summary ). Please make sure your title follow these conventions -->

## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Fixes #(5375)
https://www.loom.com/share/dac18479747c483cbe76128d8526967f?sid=15e49bc7-1c5d-4bb4-87ec-74354d062a1c


<!-- Please provide a screenshots or a loom video for visual changes to speed up reviews
 Loom Video: https://www.loom.com/
-->

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

- Go to a survey summary with >0 responses
- Switch to the responses tab
- Select >0 rows in the response table
- Interact with the context bar

## Checklist

<!-- We're starting to get more and more contributions. Please help us making this efficient for all of us and go through this checklist. Please tick off what you did  -->

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

victorvhs017 commented on apps/web/modules/ui/components/data-table/components/data-table-toolbar.tsx:
Can we add a test file for this one?

victorvhs017 commented on apps/web/modules/ui/components/data-table/components/selected-row-settings.test.tsx:
```diff
@@ -0,0 +1,188 @@
+    vi.mocked(toast.success).mockClear();
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+  // cleanup DOM after each testcle
+  afterEach(() => cleanup());
+
```
@jakobsitory we should fix this one

victorvhs017 reviewed (changes requested):
Great work!! 💪 

Can you add the missing test file and also solve the issue that was flagged by the SonarQube check?

https://sonarcloud.io/project/issues?id=formbricks_formbricks&pullRequest=5488&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true

I'll do a full review after these changes

jakobsitory commented:
@victorvhs017: I added the test file and fixed the other sonarQube issue
@jobenjada: Included a fix that prevents reload on download -> added the fix in the button to download all responses as well to stay consistent

jobenjada reviewed (approved):
looks good, thanks for the tweaks! :)

victorvhs017 commented on uploads/ars2tjk8hsi8oqk1uac00mo8/private/export-mock_survey-2025-05-06-16-24-24.csv:
🔧 **Fix**: This should not be committed.

jakobsitory replied:
✅

victorvhs017 commented on uploads/ars2tjk8hsi8oqk1uac00mo8/private/export-mock_survey-2025-05-06-16-24-24.xlsx:
🔧 **Fix**: This should not be committed.

jakobsitory replied:
✅

victorvhs017 reviewed (changes requested):
❓ **Question**: Is the download only available for responses? Should it be available for contacts, too? The component used is the same.

jakobsitory commented on apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/responses/components/ResponseTable.tsx:
```diff
@@ -180,6 +182,33 @@ export const ResponseTable = ({
+        toast.error(errorMessage || t("environments.surveys.responses.error_downloading_responses"));
+      }
+    } catch (error) {
+      console.error(error);
```
fixed in next commit

jakobsitory replied:
✅

jakobsitory commented on apps/web/modules/ui/components/data-table/components/data-table-toolbar.tsx:
```diff
@@ -27,14 +28,21 @@ export const DataTableToolbar = <T,>({
+          deleteRows={deleteRows}
+          type={type}
+          deleteAction={deleteAction}
+          downloadRows={downloadRows}
```
Fair.
Changed it that way:

deleteRowsAction -> function that deletes multiple rows
deleteAction -> function that deletes one row
downloadRowsAction -> function that downloads multiple rows
